spec/UserMiddleware.coffee

```diff
@@ -0,0 +1,258 @@
+noflo = require 'noflo'
+
+unless noflo.isBrowser()
+  chai = require 'chai'
+  path = require 'path'
+  baseDir = path.resolve __dirname, '../'
+else
+  baseDir = 'noflo-ui'
+
+describe 'User Middleware', ->
+  c = null
+  actionIn = null
+  passAction = null
+  newAction = null
+  before (done) ->
+    @timeout 4000
+    loader = new noflo.ComponentLoader baseDir
+    loader.load 'ui/UserMiddleware', (err, instance) ->
+      return done err if err
+      c = instance
+      actionIn = noflo.internalSocket.createSocket()
+      c.inPorts.action.attach actionIn
+      actionIn.port = 'action'
+      ###
+      c.network.on 'begingroup', (data) ->
+        console.log "   < #{data.id}"
```

Comment:
No flowtraces on bare (non-runtime) NoFlo networks I think? Would be super useful